### PR TITLE
feat: adds simple loader

### PR DESCRIPTION
**Pull request by ascorbic**
## Changes

Adds a simplified syntax for loaders, for cases where no granular updates or delta sync is needed. The loader returns an array of objects, which regenerates the collection from scratch, replacing the existing content. This is meant to be available in addition to the lower level loader API, but I don't know how best to explain this, or if this is the right approach so would welcome feedback. This is the API as implemented in this PR:

```ts
const cats = defineCollection({
	type: "experimental_data",
	loader: async function() {
		return  [{
				"breed": "Siamese",
				"id": "siamese",
				"size": "Medium",
				"origin": "Thailand",
				"lifespan": "15 years",
				"temperament": ["Active", "Affectionate", "Social", "Playful"]
			},
			 {
				"breed": "Persian",
				"id": "persian",
				"size": "Medium",
				"origin": "Iran",
				"lifespan": "15 years",
				"temperament": ["Calm", "Affectionate", "Social"]
			},
			 {
				"breed": "Tabby",
				"id": "tabby",
				"size": "Medium",
				"origin": "Egypt",
				"lifespan": "15 years",
				"temperament": ["Curious", "Playful", "Independent"]
			},
			{
				"breed": "Ragdoll",
				"id": "ragdoll",
				"size": "Medium",
				"origin": "United States",
				"lifespan": "15 years",
				"temperament": ["Calm", "Affectionate", "Social"]
			}
		];
	},
	schema: z.object({
		breed: z.string(),
		id: z.string(),
		size: z.string(),
		origin: z.string(),
		lifespan: z.string(),
		temperament: z.array(z.string())
	}),
})
```

You can compare it with the lower-level loader API here: https://github.com/withastro/astro/blob/content-layer/packages/astro/test/fixtures/content-layer/src/loaders/post-loader.ts

## Testing

Adds a test and updates the fixture

### CODE REVIEW

File: packages/astro/test/fixtures/content-layer/src/content/config.ts [hunk 1]
```diff
@@ -18,4 +18,52 @@ const dogs = defineCollection({
 		temperament: z.array(z.string())
 	}),
 })
-export const collections = { blog, dogs };
+
+const cats = defineCollection({
+	type: "experimental_data",
+	loader: async function() {
```
Comment (ematipico) [inline packages/astro/test/fixtures/content-layer/src/content/config.ts:24]: Why do we define `async` if the function doesn't return any promise?

File: packages/astro/src/content/loaders.ts [hunk 2]
```diff
@@ -16,6 +16,11 @@ export interface ParseDataOptions {
 	filePath?: string;
 }
 
+export type DataWithId = {
+	id: string;
+	[key: string]: unknown;
```
Comment (ematipico) [inline packages/astro/src/content/loaders.ts:21]: Here, we define the data as `unknown`, but the zod schema is `z.any`. Is it intentional?

File: packages/astro/test/fixtures/content-layer/src/content/config.ts [hunk 3]
```diff
@@ -18,4 +18,52 @@ const dogs = defineCollection({
...
+				"breed": "Ragdoll",
+				"id": "ragdoll",
+				"size": "Medium",
+				"origin": "United States",
+				"lifespan": "15 years",
+				"temperament": ["Calm", "Affectionate", "Social"]
+			}
+		];
+	},
+	schema: z.object({
```
Comment (ematipico) [inline packages/astro/test/fixtures/content-layer/src/content/config.ts:59]: The schema shapes an entry, and the loader returns an array. I wonder why this is the case. Wouldn't it make more sense to match the schema with the data returned by the loader? Why are we forced to return an array and not a single object?

File: packages/astro/test/fixtures/content-layer/src/content/config.ts (hunk 1 above)
Comment (ascorbic) [reply packages/astro/test/fixtures/content-layer/src/content/config.ts:24]: Just so the fixture tests that promises work

File: packages/astro/test/fixtures/content-layer/src/content/config.ts (hunk 3 above)
Comment (ascorbic) [reply packages/astro/test/fixtures/content-layer/src/content/config.ts:59]: The schema is for each item though. The array is for the whole collection

File: packages/astro/src/content/loaders.ts (hunk 2 above)
Comment (ascorbic) [reply packages/astro/src/content/loaders.ts:21]: I've tightened up the schema

File: packages/astro/test/fixtures/content-layer/src/content/config.ts (hunk 1 above)
Comment (ematipico) [reply packages/astro/test/fixtures/content-layer/src/content/config.ts:24]: Do you know if it is mandatory? I saw that the types accept returning data that don't return a promise

File: packages/astro/test/fixtures/content-layer/src/content/config.ts (hunk 3 above)
Comment (ascorbic) [reply packages/astro/test/fixtures/content-layer/src/content/config.ts:59]: Do you mean returning the collection as an object with the ids as keys, or something else?

File: packages/astro/test/fixtures/content-layer/src/content/config.ts (hunk 1 above)
Comment (ascorbic) [reply packages/astro/test/fixtures/content-layer/src/content/config.ts:24]: We `await` it [here](https://github.com/withastro/astro/pull/11386/files#diff-fff398c3bcb68ee6e2e9161735922902d26b20550bf701ca1dc745ce7baaef79R151), so it's supported but not mandatory.

File: packages/astro/test/fixtures/content-layer/src/content/config.ts (hunk 3 above)
Comment (ematipico) [reply packages/astro/test/fixtures/content-layer/src/content/config.ts:59]: IMHO, if a `schema` defines an object, I would expect the `loader` to return an object. If a `schema` defines an array, I would expect the `loader` to return an array.

However, if we decide to go with the current design, I think we should rename `schema` and `loader`, for example, `entrySchema` or `entitySchema`, and `entitiesLoader` or `entriesLoader`. 

Anyway, not a blocker, we can merge this PR any time

**Review by ematipico — APPROVED** [review]

File: packages/astro/test/fixtures/content-layer/src/content/config.ts (hunk 3 above)
Comment (ascorbic) [reply packages/astro/test/fixtures/content-layer/src/content/config.ts:59]: This is how collections work now though. The schema is for the individual items, not the whole collection array

**Review by bluwy — APPROVED** [review]

**Review by matthewp — APPROVED** [review]

**Pull request merged**